relay/sources/rudder-pkg/lib/rudder-pkg/rpkg.py

```diff
@@ -0,0 +1,151 @@
+import os, logging, re, json
+from pkg_resources import parse_version
+
+import rudderPkgUtils as utils
+
+"""
+    Versions can contains the words "-SNAPSHOT" at the end, this should not be a problem at the moment since it is only present in nightly.
+    Moreover, we should not have to compare a nightly version to a release one.
+"""
+class Version:
+    def __init__(self, pluginLongVersion):
+        match = re.search(r'(?P<rudderVersion>[0-9]+\.[0-9]+)-(?P<pluginShortVersion>[0-9]+\.[0-9]+(-(?P<mode>[a-zA-Z]+)){0,1})', pluginLongVersion)
+        if match.group('mode') is None:
+            self.mode = 'release'
+        elif match.group('mode') == 'SNAPSHOT':
+            self.mode = 'nightly'
+        else:
+            utils.fail("The version %s does not respect the version syntax"%(pluginLongVersion))
+
+        if match.group('rudderVersion') is None or match.group('pluginShortVersion') is None:
+            utils.fail("The version %s does not respect the version syntax [0-9]+.[0-9]+-[0-9]+.[0-9]+(-SNAPSHOT){0,1}"%(pluginLongVersion))
+        else:
+            self.rudderVersion = match.group('rudderVersion')
+            self.pluginShortVersion = match.group('pluginShortVersion')
+            self.pluginLongVersion = pluginLongVersion
+
+    def __str__(self):
+        return self.mode + " => " + self.rudderVersion + "-" + self.pluginShortVersion + " | " + self.pluginLongVersion
```

Comment:
i think what is after | is redundant for __str__ (but it can still be used in __repr__ in case of bug)

Reply:
Anyway, I'm pretty sure I'm not using the __str__ anywhere in the code anymore, more here to easy debug than anything else